### ISSUE TITLE
WebExtension API for Safari 14.0

### ISSUE THREAD

**Issue by UInt2048** [Tampermonkey/tampermonkey#1031]
## Expected Behavior
TamperMonkey for Safari should support the cross-platform [WebExtension API](https://developer.apple.com/documentation/safariservices/safari_web_extensions) used in other browsers' TamperMonkey versions introduced in Safari 14.0 (which supports macOS 10.14+)

## Actual Behavior
There is only a $1.99 version of TamperMonkey for Safari 13+ which does not use it.

## Specifications

- Chrome: 87.0.4251.0 (Canary channel)
- Safari: 14.0 (15610.1.28.1.9, 15610)
- TM: 4.11.6119
- OS: macOS 10.15.6

## Script

Not applicable

**Comment by UInt2048** [comment]
@derjanb Do you plan to support this in Safari or should I close this? I'd love to know what your plan is...

**Comment by derjanb** [comment]
Yes, but I'm not sure exactly when this will happen because again it's not just about repackaging a WebExtension and you're done, but as I found out after a first brief investigation, there are a lot of modifications required to adapt a WebExtension for Safari 14.

Also the current solution supports Safari 12, 13 and 14 as well.

**Comment by zhiyuanzhai** [comment]
So, is there any plan for this?

**Comment by derjanb** [comment]
Tampermonkey based on web extension API is available now...

I'm not entirely sure if there is an issue in the macOS version. I got everything working on my end. Especially `GM_notification` was difficult to get working in iOS, but it's working now.
So I worked on renaming the Safari App based Tampermonkey version to "Tampermonkey Classic", uploaded and published it.
Back at work on the WebExt version, notifications stopped working with

`"UNErrorDomain" error 1, the operation couldn't be completed"

I think this is maybe related to renaming the app, which got the notification permission long time ago, but all tried [ways to make macOS forget the extension](https://stackoverflow.com/questions/70039047/unable-to-request-authorization-for-local-notifications-on-macos) didn't work. 

Could someone do me a favor and try the macOS version, and tell me if notifications via `GM_notification` are working for you?
